### ISSUE TITLE
Are you still working on this?

### ISSUE THREAD

**Issue by ym1234** [polytomous/TerrariaClone#7]

**Comment by polytomous** [comment]
I haven't been working on it lately, but I certainly haven't stopped thinking about this. Would you like to do some work on this? I can give you pointers and we can pair program if you'd like. 

In general, I'm to the point where I want to cross port in the other patches that fix the rendering issues on the game. Once that is done, it's to the point where you can do serious playtesting, and there's a universe of unit tests to write here.

**Comment by ym1234** [comment]
I would love to work with you, but I haven't worked with the java2d api at all, and also I kinda wanted to tackle this challenge alone (I haven't read the spoilers).

I justed wanted to check if you're still working on it, Thanks for suggesting though.

**Comment by polytomous** [comment]
AWESOME. I'll make sure not to spoil you on anything. Hopefully the issue titles on this repository haven't ruined your fun already. 👍 This is a super fun (and horrific oh my god so horrific) project to toy around with. 

If you find any cool bugs, I'd love it if you'd at least leave repro steps for them so I can try to fix them on my fork. And if you do need any help, I'd love to help out. 

Let me know if you need anything.

**Comment by polytomous** [comment]
@ym1234 I have a burning urge to encourage you to work on this code somehow. But *this* code isn't _this_ code. I need to make a new branch, starting back at the root of this experiment, that will faithfully build (though maybe not run) on at least 2 different ides. We can of course, do the same thing to the tip of this code too, but my code is broken in other profound ways and has too many spoilers for true TerrariaClone fans. 

That really means just a whole bunch of configuration work. I will probably open a distinct issue to describe this work, but i have other High Priority Sort Of Things to work on for the next short term of time. 

Ideally, if this first pass through the code was a hellscape, the next version has to follow through with the Dante Divine Comedy reference and it would need to be purgatory, with a bit more of a nicer foundation to work from, and hopefully a whole host of Virgils that could guide you. I started on the least important part of that work (skimming through Purgatorio to find some dope quotes) but I would love for you to have the opportunity to work on the most important parts.

I can give you more details and specifics / guidance. This is just a reopening salvo. I answered you wrong. I should have told you that the java2d stuff isn't the most important bit. You are. There is a way you can read through this code and make fixes that will bring it into a language you understand, and then provide those fixes so that everything gets better. 

I have a bunch more writing and working to do on this project but I would *love* the absolute privilege of having you alongside me.

**Comment by polytomous** [comment]
@ym1234 You have to know this about me: 

```
"I was born with small access to big dreams."
```

```
"A slight high from being able to buy
What you shouldn't but you never knew why
Never mind, this the feeling I've been wanting for time
Flying blind, into a wall strong drawing to shine"

"I just looked at the pictures
Didn't have a manual to build this
Just assemble it from feeling
By looking at pictures
I was given tools meant for building
With no instruction for the wielding
But look at the picture
It ain't even close to the description
And yet it's working as I wished it
By looking at pictures
And even though parts of it are missing
It's still complete in the depiction"
```

- Built By Pictures - Oddisee
https://youtu.be/SPp-ptqXHIA

**Comment by polytomous** [comment]
https://gist.github.com/polytomous/532a1404fa8716ce4078b90b5051dff7   also just gonna stash this here. ;) umapathy 17 is bae <3

**Comment by ym1234** [comment]
The main problem is lack of time right now as I'm starting my final year in high school (which is the most important year where I live) and working on some other side hobbies, so if you don't mind me not "giving my all" on this, I'd love to work with you.

My discord is ym555#1339, since discussion on Github issues is not the best :P

**Comment by polytomous** [comment]
I don't want to steal your time. Spend what you must to keep yourself safe so that you can continue to wager and flip coins. 
I can't own those stars that you are building in the nebula of your schooling. 
But with whatever small, second order of smallness that you can spare, plausibly derivative in the grand scheme of things, I would hope you could figure out how to place yourself in the right places where you could multiply folks by ten. 

Maybe that place isn't here right now. But I would hope that I would win the lottery to earn the privilege to work with you someday when your time is more abundant, and your future just as glistening as it is now.  

I know how I need to work. But not entirely. And maybe there is a universe of work I need to do before you take the first efforts. 

Opportunity costs are real. Maybe there is some way that we can cross apply this effort you would "waste" knowing that you hadn't wasted it all. 

Here is a low cost strategy for you to contribute. Make some noise in the issue tracker with what I need to fix. Only spend an hour. Open as many issues as you like. And if you had a dream that you wanted built, put it in the issue tracker so someone can pick up that work. 

I'll do what I can. 

The most important way that we can keep terrariaclone from dying is to make it so so so so so so easy just to get it built. That means that we have to think about the hardest, most frustrating part of any project: environment setup. getting all of the tools ready and documenting that they can work. 

And you told me earlier that you were worried you didn't know enough. Your blank mind (more than, I know you've already contributed fixes) could be enough for us to travel back in time on this merkle dag to start a new place and a new story. 

Maybe I just want you to brain storm. Share with me your dream. And I will find a way with the time that you cannot spend, that i have earned to relax, to mix it into the next version. A version I did alone is a failure. 

I need to not fail again. TerrariaClone deserves my best. U deserve the right to tell me what I must do such that when you have time, I will be so ready for you. And if I can be ready for you when you are ready, then I will have also fed 2 spiders with one juicy mosquito. 

Your potentiality is limitless. Just file some issues on the parts that are hard, and I will endeavor to fix them by the time you are ready.
